enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx

```diff
@@ -1,32 +1,44 @@
-import type { ReactNode, JSX } from "react";
+import { type ReactNode, type JSX, useEffect } from "react";
 import { memo } from "react";
 import { Provider } from "react-redux";
 
 import { AppInitializeController } from "embedding-sdk/components/private/AppInitializeController";
+import type { SdkPluginsConfig } from "embedding-sdk/lib/plugins";
 import { store } from "embedding-sdk/store";
+import { setPlugins } from "embedding-sdk/store/reducer";
 import type { SDKConfigType } from "embedding-sdk/types";
 import { EmotionCacheProvider } from "metabase/styled-components/components/EmotionCacheProvider";
 import { ThemeProvider } from "metabase/ui/components/theme/ThemeProvider";
 
 import "metabase/css/vendor.css";
 import "metabase/css/index.module.css";
 
+interface MetabaseProviderProps {
+  children: ReactNode;
+  config: SDKConfigType;
+  pluginsConfig?: SdkPluginsConfig;
+}
+
 const MetabaseProviderInternal = ({
   children,
   config,
-}: {
-  children: ReactNode;
-  config: SDKConfigType;
-}): JSX.Element => (
-  <Provider store={store}>
-    <EmotionCacheProvider>
-      <ThemeProvider>
-        <AppInitializeController config={config}>
-          {children}
-        </AppInitializeController>
-      </ThemeProvider>
-    </EmotionCacheProvider>
-  </Provider>
-);
+  pluginsConfig,
+}: MetabaseProviderProps): JSX.Element => {
+  useEffect(() => {
+    store.dispatch(setPlugins(pluginsConfig || null));
```

Comment:
Does this trigger component rerender if `pluginConfig` changes? It's common for users to not memoize the config and pass the whole config object and that would change every render.